dbms/src/Storages/TrivialBuffer.h

```diff
@@ -0,0 +1,147 @@
+#pragma once
+
+#include <mutex>
+#include <thread>
+#include <ext/shared_ptr_helper.hpp>
+#include <Core/NamesAndTypes.h>
+#include <Storages/IStorage.h>
+#include <DataStreams/IBlockOutputStream.h>
+#include <Poco/Event.h>
+
+namespace Poco { class Logger; }
+
+namespace DB
+{
+
+class Context;
+
+/** При вставке буферизует входящие блоки, пока не превышены некоторые пороги.
+  * Когда пороги превышены - отправляет блоки в другую таблицу в том же порядке,
+  * в котором они пришли в данную таблицу.
+  *
+  * Пороги проверяются при вставке, а также, периодически, в фоновом потоке
+  * (чтобы реализовать пороги по времени).
+  * Если в таблицу вставляется блок, который сам по себе превышает max-пороги, то он
+  * записывается сразу в подчинённую таблицу без буферизации.
+  * Пороги могут быть превышены. Например, если max_rows = 1 000 000, в буфере уже было
+  * 500 000 строк, и добавляется кусок из 800 000 строк, то в буфере окажется 1 300 000 строк,
+  * и затем такой блок будет записан в подчинённую таблицу
+  *
+  * При уничтожении таблицы типа TrivialBuffer и при завершении работы, все данные сбрасываются.
+  * Данные в буфере не реплицируются, не логгируются на диск, не индексируются. При грубом
+  * перезапуске сервера, данные пропадают.
+  */
+class TrivialBuffer : private ext::shared_ptr_helper<TrivialBuffer>, public IStorage
```

Comment:
Need to implement base class for StorageTrivialBuffer and StorageBuffer with common code to avoid copy-paste.